common/zoneops.go

```diff
@@ -42,38 +45,31 @@ VALUES (?, ?, ?, datetime('now'), ?, ?)`
 	stmt, err := mdb.Prepare(AZsql)
 	if err != nil {
 		log.Printf("Error in SQL prepare(%s): %v", AZsql, err)
+		return fmt.Sprintf("Error in SQL prepare(%s): %v", AZsql, err), err
 	}
 
-	// mdb.mu.Lock()
-//	tx, err := mdb.Begin()
-//	if err != nil {
-//		log.Printf("AddZone: Error from mdb.Begin(): %v", err)
-//	}
-	// defer tx.Commit()
-	
 	_, err = stmt.Exec(fqdn, z.ZoneType, "", "", z.FSMMode)
 	if CheckSQLError("AddZone", AZsql, err, false) {
-		// mdb.mu.Unlock()
-		// tx.Rollback()
-		return err, ""
+		return "", err
 	}
-	// mdb.mu.Unlock()
-	// tx.Commit()
 
 	if group != "" {
 		fmt.Printf("AddGroup: the zone %s has the signergroup %s specified so we set that too\n", z.Name, group)
-		dbzone, _ := mdb.GetZone(tx, z.Name)
-		err, _ := mdb.ZoneJoinGroup(tx, dbzone, group, enginecheck) // we know that the zone exist
+		dbzone, _, err := mdb.GetZone(tx, z.Name)
+		if err != nil {
+		   return "", err
+		}
+		_, err = mdb.ZoneJoinGroup(tx, dbzone, group, enginecheck) // we know that the zone exist
 		if err != nil {
-			return err, fmt.Sprintf(
-				"Zone %s was added, but failed to attach to signer group %s.", fqdn, group)
+			return fmt.Sprintf(
+				"Zone %s was added, but failed to attach to signer group %s.", fqdn, group), err
 		} else {
-			return nil, fmt.Sprintf(
-				"Zone %s was added and immediately attached to signer group %s.", fqdn, group)
+			return fmt.Sprintf(
+				"Zone %s was added and immediately attached to signer group %s.", fqdn, group), err
```

Comment:
If you know the error is nil you usually return nil instead of the err variable to make it clear that you know it's nil.

Reply:
Sant.